topostats/grains.py

```diff
@@ -2443,3 +2379,72 @@ def remove_disconnected_grains(
                     predicted_grain_tensor[predicted_mask_region_mask, channel] = 0
 
         return predicted_grain_tensor
+
+
+# pylint: disable=too-many-branches
+def get_grain_thresholds(  # noqa: C901
+    image: npt.NDArray,
+    threshold_method: str,
+    otsu_threshold_multiplier: float | None = None,
+    threshold_std_dev: list | None = None,
+    absolute: list | None = None,
+) -> list[float]:
+    """
+    Obtain thresholds for masking data points.
+
+    Parameters
+    ----------
+    image : npt.NDArray
+        2D Numpy array of image to be masked.
+    threshold_method : str
+        Method for thresholding, 'otsu', 'std_dev' or 'absolute' are valid options.
+    otsu_threshold_multiplier : float
+        Scaling value for Otsu threshold.
+    threshold_std_dev : dict
+        Dict of above and below thresholds for the standard deviation method.
+    absolute : tuple
+        Dict of below and above thresholds.
+
+    Returns
+    -------
+    dict[str, list[float]]
+        Dictionary of thresholds, contains keys 'below' and optionally 'above'.
+    """
+    thresholds: list[float] = []
```

Comment:
Shouldn't need to create and assign to `thresholds` can just return the values directly (see comments below).